modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java

```diff
@@ -2267,6 +2268,70 @@ public RpcMsg ops_getTransaction(Object _params) {
         return new RpcMsg(result);
     }
 
+    public RpcMsg ops_getBlockDetails(Object _params){
+        String _bnOrHash;
+        if (_params instanceof JSONArray) {
+            _bnOrHash = ((JSONArray) _params).get(0) + "";
+        } else if (_params instanceof JSONObject) {
+            _bnOrHash = ((JSONObject) _params).get("block") + "";
+        } else {
+            return new RpcMsg(null, RpcError.INVALID_PARAMS, "Invalid parameters");
+        }
+        AionBlock block = null;
+        Long bn = this.parseBnOrId(_bnOrHash);
+
+        // user passed a Long block number
+        if (bn != null) {
+            if (bn >= 0) {
+                block = (AionBlock) this.ac.getBlockchain().getBlockByNumber(bn);
+                if (block == null) {
+                    return new RpcMsg(JSONObject.NULL);
+                }
+            } else {
+                return new RpcMsg(JSONObject.NULL);
+            }
+        }
+
+        // see if the user passed in a hash
+        if (block == null) {
+            block = (AionBlock) this.ac.getBlockchain().getBlockByHash(ByteUtil.hexStringToBytes(_bnOrHash));
+            if (block == null) {
+                return new RpcMsg(JSONObject.NULL);
+            }
+        }
+
+        Block mainBlock = this.ac.getBlockchain().getBlockByNumber(block.getNumber());
+        if (mainBlock == null) {
+            return new RpcMsg(JSONObject.NULL);
+        }
+
+        if (!Arrays.equals(block.getHash(), mainBlock.getHash())) {
+            return new RpcMsg(JSONObject.NULL);
+        }
+        Block previousBlock = this.ac.getBlockchain().getBlockByHash(block.getParentHash());
+
+        BigInteger blkReward =
+            ((AionBlockchainImpl) ac.getBlockchain())
+                .getChainConfiguration()
+                .getRewardsCalculator()
+                .calculateReward(block.getHeader().getNumber());
+        BigInteger totalDiff =
+            this.ac.getAionHub().getBlockStore().getTotalDifficultyForHash(block.getHash());
+
+        List<AionTxInfo> txInfoList = new ArrayList<>();
+        for (AionTransaction transaction: block.getTransactionsList()){
+            AionTxInfo txInfo = this.ac.getAionHub()
+                .getBlockchain()
+                .getTransactionInfo(transaction.getTransactionHash());
+            txInfoList.add(txInfo);
+        }
+
+
+        return new RpcMsg(
+            Blk.aionBlockDetailsToJson(block,
+                txInfoList, previousBlock==null ? null:previousBlock.getTimestamp(), totalDiff, blkReward));
```

Comment:
what causes previousBlock == null?

Reply:
This case occurs if the requested block is the genesis block.